projects/hathor-wallet-headless/0004-tx-transaction.md

```diff
@@ -50,298 +51,300 @@ flowchart TD;
+  token: TemplateRef.or(z.string().regex(/^[a-fA-F0-9]{64}|00$/).default('00')),
+  address: TemplateRef.or(z.string()).optional(),
+  timelock: TemplateRef.or(z.coerce.number()).optional(),
+  checkAddress: z.boolean().optional(),
+});
 ```
 
+- `position`: which index to insert the input, default to `-1`
+  - can be `-1` to insert at the end
+  - the request will fail if position doesn't exists
 - `amount`: amount of tokens
 - `token`: which token to select, defaults to native token, in practice `00` (HTR). User must set the token UID here.
 - `address`: create an output script for the address, if not present will get one from the wallet
-  - [NOTE-1]: It will get a not used address from the wallet
-- `authority`: `mint` or `melt`, if present will create the desired authority output
-  - [NOTE-2]: If the required input is not present, then it will generate an invalid transaction
+  - It will get a not used address from the wallet
 - `timelock`: UNIX timestamp, the generated output may not be spent before this date and time.
 - `checkAddress`: whether to check that the address is from the wallet, defaults to `false`
-  - [NOTE-3]: If `true` and address is not from the wallet, then the interpreter should fail short
+  - If `true` and address is not from the wallet, then the interpreter should fail
 
 >[!NOTE]
-> Decision 3: Should we keep the default behavior on `address` or make it more strict? Like, if there isn't any `address` value, then fail short.
+> Decision: Should we keep the default behavior on `address` or make it more strict? Like, if there isn't any `address` value, then fail short.
 >
 > To counter the behavior of a stricter `address` we can add a flag to let the user choose the desired behavior, either get a new address or get an address by index.
 >
 > Or we can make `address` a poli typed property which can accept `number` for indexes, the string `new` or a general `string` to denote an address.
 
-#### `DataOutputTemplateInstruction`
+#### `AuthorityOutputInstruction`
 
-type: `output/data`
+Add an authority output to the transaction
 
 ```ts
-interface DataOutputTemplateInstruction extends OutputTemplateInstruction {
-  data: string;
-}
+const AuthorityOutputInstruction = z.object({
+  type: z.literal('output/authority'),
+  position: z.number().default(-1),
+  count: TemplateRef.or(z.coerce.number()),
+  token: TemplateRef.or(z.string().regex(/^[a-fA-F0-9]{64}$/)),
+  authority: z.enum(['mint', 'melt']),
+  address: TemplateRef.or(z.string()).optional(),
+  timelock: TemplateRef.or(z.coerce.number()).optional(),
+  checkAddress: z.boolean().optional(),
+});
 ```
 
-- `data`: UTF-8 encoded string of data
+- `position`: which index to insert the input, default to `-1`
+  - can be `-1` to insert at the end
+  - the request will fail if position doesn't exists
+- `count`: amount of authority outputs to create.
+- `token`: which token to select, defaults to native token. User must set the token UID here.
```

Comment:
question: Would it be possible to create an authority for the HTR token?

Reply:
Fixed